### DISCUSSION TITLE
Not all gradle dependencies are found and analysed

### DISCUSSION THREAD

**Discussion by siebenth** [renovatebot/renovate#22948] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:35.140.0-slim

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Not all dependencies in our gradle (kotlin dsl based) are found and searched for updates.
From the gradle plugin dependencies in `build.gradle.kts`
```
buildscript {
    dependencies {
        classpath("org.springframework.boot:spring-boot-gradle-plugin:3.0.6")
//		 classpath("org.springframework.boot:spring-boot-gradle-plugin:${ModuleConfiguration.Build.Plugin.SpringBoot.pluginVersion}")
//		 classpath("com.netflix.nebula:nebula-release-plugin:16.0.0")
		 classpath("com.netflix.nebula:nebula-release-plugin:${ModuleConfiguration.Build.Plugin.NebulaRelease.version}")
    }
}
```
the first (`classpath("org.springframework.boot:spring-boot-gradle-plugin:3.0.6")`) is found, analysed and leads to a pull request. The second (`classpath("com.netflix.nebula:nebula-release-plugin:${ModuleConfiguration.Build.Plugin.NebulaRelease.version}")`) with the indirection into `ModuleConfiguration.kt` in `buildSrc` is ignored.
Normal dependencies like `runtimeOnly("com.h2database:h2:${ModuleConfiguration.Build.Database.h2Version}")` are found and handled correctly.
As workaround we could move all versions strings into the `*.gradle.kts` files but we'd like to have them central in `buildSrc`.
A minimalistic gradle project can be found under https://github.com/siebenth/renovate-gradle-issue. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by Churro** [comment, marked as answer]
`ModuleConfiguration.Build.Plugin.NebulaRelease.version` is located in the "4th hierarchy level". The parser currently only goes down three nesting layers but your sample confirmed that it's reasonable to increase this. I'll prepare a PR.

**Reply by siebenth** [reply]
Thank you very much for the quick resolution. With the version 35.141.3-slim all those dependencies are now found and result in PRs!
